src/io/flutter/run/OpenDevToolsAction.java

```diff
@@ -57,30 +60,23 @@ public void actionPerformed(@NotNull final AnActionEvent event) {
       return;
     }
 
-    final DevToolsManager devToolsManager = DevToolsManager.getInstance(project);
 
-    if (myConnector == null) {
-      if (devToolsManager.hasInstalledDevTools()) {
-        devToolsManager.openBrowser(myApp);
-      }
-      else {
-        final CompletableFuture<Boolean> result = devToolsManager.installDevTools();
-        result.thenAccept(o -> devToolsManager.openBrowser(myApp));
-      }
-    }
-    else {
-      final String urlString = myConnector.getBrowserUrl();
-      if (urlString == null) {
+    AsyncUtils.whenCompleteUiThread(DevToolsService.getInstance(project).getDevToolsInstance(), (instance, ex) -> {
+      if (ex != null) {
+        LOG.error(ex);
         return;
       }
 
-      if (devToolsManager.hasInstalledDevTools()) {
-        devToolsManager.openBrowserAndConnect(myApp, urlString);
-      }
-      else {
-        final CompletableFuture<Boolean> result = devToolsManager.installDevTools();
-        result.thenAccept(o -> devToolsManager.openBrowserAndConnect(myApp, urlString));
+      if (project == null) {
```

Comment:
you also need to check whether the project has been disposed. I don't think checking whether the project is null does much.